receiver/tlscheckreceiver/scraper.go

```diff
@@ -30,6 +36,70 @@ type scraper struct {
 	getConnectionState func(endpoint string) (tls.ConnectionState, error)
 }
 
+// listens to the error channel and combines errors sent from different go routines,
+// returning the combined error list should context timeout or a nil error value is
+// sent in the channel signifying the end of a scrape cycle
+func errorListener(ctx context.Context, eQueue <-chan error, eOut chan<- *scrapererror.ScrapeErrors) {
+	errs := &scrapererror.ScrapeErrors{}
+
+	for {
+		select {
+		case <-ctx.Done():
+			eOut <- errs
+			return
+		case err, ok := <-eQueue:
+			if !ok {
+				eOut <- errs
+				return
+			}
+			errs.AddPartial(1, err)
+		}
+	}
+}
+
+func validatePort(port string) error {
+	portNum, err := strconv.Atoi(port)
+	if err != nil {
+		return fmt.Errorf("provided port is not a number: %s", port)
+	}
+	if portNum < 1 || portNum > 65535 {
+		return fmt.Errorf("provided port is out of valid range (1-65535): %d", portNum)
```

Comment:
This would be `[1, 65535]` if using set notation right?

Reply:
Good point, I updated the message to clarify the inclusive bounds using set notation.